lib/Dialect/SV/SVOps.cpp

```diff
@@ -1944,6 +1946,318 @@ void BindOp::build(OpBuilder &builder, OperationState &odsState, StringAttr mod,
+  if (!portLocs.empty())
+    result.addAttribute("port_locs", builder.getArrayAttr(portLocs));
+
+  // parse verbatim content
+  if (!result.attributes.get("content")) {
+    return parser.emitError(parser.getCurrentLocation(),
+                            "sv.verbatim.module requires 'content' attribute");
+  }
+
+  // parse output file
+  auto outputFileAttr = result.attributes.get("output_file");
+  if (!outputFileAttr) {
+    return parser.emitError(
+        parser.getCurrentLocation(),
+        "sv.verbatim.module requires 'output_file' attribute");
+  }
+  if (!isa<hw::OutputFileAttr>(outputFileAttr)) {
+    return parser.emitError(
+        parser.getCurrentLocation(),
+        "sv.verbatim.module 'output_file' attribute must be an OutputFileAttr");
+  }
+
+  return success();
+}
+
+void SVVerbatimModuleOp::setAllPortNames(ArrayRef<Attribute> names) {
+  auto moduleType = getModuleType();
+  assert(names.size() == moduleType.getNumPorts() &&
+         "Number of names must match number of ports");
+
+  SmallVector<hw::ModulePort> newPorts;
+  for (auto [i, port] : llvm::enumerate(moduleType.getPorts())) {
+    auto newName = cast<StringAttr>(names[i]);
+    newPorts.push_back({newName, port.type, port.dir});
+  }
+
+  auto newModuleType = hw::ModuleType::get(getContext(), newPorts);
+  setModuleTypeAttr(TypeAttr::get(newModuleType));
+}
+
+LogicalResult SVVerbatimModuleOp::verify() {
+  auto moduleType = getModuleType();
+  if (!moduleType)
+    return emitOpError("missing module type");
+
+  // must have verbatim content
+  if (getContent().empty())
+    return emitOpError("missing or empty content attribute");
+
+  auto numPorts = moduleType.getNumPorts();
+
+  if (auto attrs = getPerPortAttrs()) {
+    if (attrs->size() != numPorts)
+      return emitOpError("port attributes array size (")
+             << attrs->size() << ") doesn't match number of ports (" << numPorts
+             << ")";
+  }
+
+  if (auto locs = getPortLocs()) {
+    if (locs->size() != numPorts)
+      return emitOpError("port locations array size (")
+             << locs->size() << ") doesn't match number of ports (" << numPorts
+             << ")";
+  }
+
+  auto outputFileAttr = getOutputFile();
+  if (!outputFileAttr || outputFileAttr.getFilename().getValue().empty())
+    return emitOpError("output_file attribute cannot be empty");
+
+  return success();
+}
+
+LogicalResult
+SVVerbatimModuleOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
+  if (auto additionalFiles = getAdditionalFiles()) {
+    auto *symtable = SymbolTable::getNearestSymbolTable(*this);
+    if (!symtable)
+      return emitError("sv.verbatim.module must exist within a region "
+                       "which has a symbol table.");
+
+    for (auto fileRef : *additionalFiles) {
+      auto symbolRef = llvm::cast<SymbolRefAttr>(fileRef);
+      auto *referencedOp = symbolTable.lookupSymbolIn(symtable, symbolRef);
```

Comment:
please remove `getnearestSymbolTable(*this)` above since it walks entire IR.
```suggestion
symbolTable.lookupNearestSymbolFrom(getOperation(), symbolRef);
```